crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -203,17 +321,14 @@ where
                     seq, e
                 ))
             });
-        let mut fn_checkpoint_guard = self
+        let fn_checkpoint_guard = self
             .metrics
-            .fullnode_checkpoint_download_latency
+            .fullnode_checkpoint_wait_and_download_latency
             .start_timer();
         while checkpoint.is_err() {
-            // sleep for 1 second and retry if latest checkpoint is not available yet
-            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-            fn_checkpoint_guard = self
-                .metrics
-                .fullnode_checkpoint_download_latency
-                .start_timer();
+            // sleep for 0.1 second and retry if latest checkpoint is not available yet
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
changed this to 0.1 sec again b/c
- this is done sequentially, and would generate 10 RPS at most
- from our read tests, getCheckpoint RPS can go very high to several thousands
- this is the starting point of whole checkpoint, so trading several light-weight RPS for potentially 0.x seconds is worth imo

Reply:
+1